packages/openapi3/src/schema-emitter.ts

```diff
@@ -645,6 +651,63 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
     }
   }
 
+  #attachXmlObject(program: Program, prop: ModelProperty, emitObject: OpenAPI3Schema) {
+    const xmlObject: OpenAPI3XmlSchema = {};
+
+    const isXmlModel =
+      prop.model && prop.model?.name !== resolveEncodedName(program, prop.model, "application/xml");
+
+    // Resolve XML name
+    const xmlName = resolveEncodedName(program, prop, "application/xml");
+    if (xmlName !== prop.name) {
```

Comment:
thtat should actually compare with the json name.
Add a test for that as well

```
@encodedName("application/json", "exp")
@encodedName("application/json", "ExpireAt")
expire: string;
```

the openapi should produce
```
name: exp
xml: 
   name: ExpireAt